src/utilities/did/did.ts

```diff
@@ -18,8 +18,29 @@ export async function getDidDocument(did: DidUri): Promise<DidDocument> {
   return details.document;
 }
 
-export function getDidEncryptionKey(details: DidDocument): DidEncryptionKey {
-  const { keyAgreement } = details;
+export function parseDidUri(did: DidUri): ReturnType<
+  typeof Utils.parseDidUri
+> & {
+  lightDid: DidUri;
+  fullDid: DidUri;
+} {
+  const parsed = Utils.parseDidUri(did);
+  const { address, type } = parsed;
+
+  const lightDid =
+    type === 'light' ? did : (`did:kilt:light:00${address}` as DidUri);
```

Comment:
This feels a bit wrong. Do we even need to reconstruct a light DID for full DIDs that have been removed from chain? Isn't this light DID useless now?

Reply:
No, when a full did is removed, you also can't use the corresponding light DID anymore.

Reply:
Currently in the full version of Sporran we're creating the light DID again when the on-chain DID is removed. `did` is a required property in the `Identity` interface. Maybe we should just make the `did` property optional and set it to undefined when the on-chain DID is removed? @arty-name what do you think?

Reply:
This might be simple in the sporran-lite, so feel free to try it out!